src/providers/wms/qgswmscapabilities.cpp

```diff
@@ -669,6 +669,44 @@ void QgsWmsCapabilities::parseCapability( QDomElement const &e, QgsWmsCapability
     }
   }
 
+  // Layers in a WMS-C Capabilities TileSet do not have layer title or abstract
+  // still those could be present in a Layer list.
+  if ( !mTileLayersSupported.isEmpty() )
+  {
+    QHash<QString, QString> titles;
+    QHash<QString, QString> abstracts;
+
+    for ( int i = 0; i < mLayersSupported.size(); ++i )
```

Comment:
Can you use the C++11 range-based loop please? In this case, it should be something like this:

``` c++
for ( const QgsWmsLayerProperty &property : qgis::as_const( mLayersSupported ) )
{
   ...
}
```

Then you may use `property` instead of `mLayersSupported.at( i )`.

Reply:
Changes have been implemented.